alacritty_terminal/src/selection.rs

```diff
@@ -165,179 +172,165 @@ impl Selection {
         }
     }
 
-    pub fn to_span<T>(&self, term: &Term<T>) -> Option<Span> {
-        // Get both sides of the selection
-        let (mut start, mut end) = match *self {
-            Selection::Simple { ref region } | Selection::Block { ref region } => {
-                (region.start.point, region.end.point)
-            },
-            Selection::Semantic { ref region } | Selection::Lines { ref region } => {
-                (region.start, region.end)
-            },
-        };
+    /// Convert selection to grid coordinates.
+    pub fn to_range<T>(&self, term: &Term<T>) -> Option<SelectionRange> {
+        let grid = term.grid();
+        let num_cols = grid.num_cols();
 
-        // Order the start/end
-        let needs_swap = Selection::points_need_swap(start, end);
-        if needs_swap {
-            std::mem::swap(&mut start, &mut end);
-        }
+        // Get selection boundaries
+        let points = self.points();
+        let (mut start, mut end) = (*points.0, *points.1);
 
-        // Clamp to visible region in grid/normal
-        let num_cols = term.dimensions().col;
-        let num_lines = term.dimensions().line.0 as isize;
-        let (start, end) = Selection::grid_clamp(start, end, num_lines, num_cols)?;
+        // Get selection sides, falling back to `Side::Left` if it will not be used
+        let sides = self.sides().unwrap_or((&Side::Left, &Side::Left));
```

Comment:
I've intentionally chosen `Left`, `Left`, to make it clear that this value is not useful. I think a `Left`/`Right` combination might make it feel more like these values actually have a specific purpose, which they do not. They exist solely to simplify the code.

Reply:
As far as I've seen, we only have combos of (Left,Right) or (Right,Left), so I've suggested it for consistency, however we can keep it this way tbh.

Reply:
> As far as I've seen, we only have combos of (Left,Right) or (Right,Left), so I've suggested it for consistency, however we can keep it this way tbh.

Yeah, left/right makes sense because that means we're talking about inclusive ranges. So it basically means that all cells are included. So maybe being different here will actually make people think twice about what that means, which I'd say is a good thing.

Also makes writing the comment easier, since just `Side::Left` has to be mentioned.